react/finger-exercises/redux/02 - react-redux/src/app/screens/App/components/ShoppingCart/index.js

```diff
@@ -1,16 +1,27 @@
-import React, { PureComponent, Fragment } from 'react';
+import React, { Component, Fragment } from 'react';
 import { arrayOf, func } from 'prop-types';
 import { bookSelectedPropType } from '@constants/propTypes';
 import Button from '@components/Button';
+import actionsCreators from '../../../../../redux/book/actions'
+import store from '@redux/store';
 
 import Item from './components/Item';
 import styles from './styles.scss';
 
-class ShoppingCart extends PureComponent {
+class ShoppingCart extends Component {
   state = {
-    open: false
+    open: false,
+    bookSelected: []
   };
 
+  componentWillMount() {
+    this.setState( {bookSelected: store.getState().bookSelected} );
+    store.subscribe(() => {
+      const { bookSelected } = store.getState();
```

Comment:
Recieve this from `mapStateToProps` and use it directly from props
https://react-redux.js.org/using-react-redux/connect-mapstate